homeassistant/components/wallbox/calendar.py

```diff
@@ -0,0 +1,104 @@
+"""Support for Wallbox calendar items."""
+
+from __future__ import annotations
+
+from datetime import datetime
+
+from homeassistant.components.calendar import (
+    CalendarEntity,
+    CalendarEntityDescription,
+    CalendarEvent,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    CHARGER_CALENDAR,
+    CHARGER_DATA_KEY,
+    CHARGER_LAST_EVENT,
+    CHARGER_SERIAL_NUMBER_KEY,
+    DOMAIN,
+)
+from .coordinator import WallboxCoordinator
+from .entity import WallboxEntity
+
+CALENDAR_TYPE = CalendarEntityDescription(
+    key=CHARGER_CALENDAR,
+    name=None,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Wallbox calendar entity."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities([WallboxCalendarEntity(coordinator, CALENDAR_TYPE)])
+
+
+class WallboxCalendarEntity(WallboxEntity, CalendarEntity):
+    """A Wallbox calendar entity."""
+
+    def __init__(
+        self, coordinator: WallboxCoordinator, description: CalendarEntityDescription
+    ) -> None:
+        """Initialize a Wallbox calendar."""
+        super().__init__(coordinator)
+        self.entity_description = description
+        self._attr_unique_id = f"{description.key}-{coordinator.data[CHARGER_DATA_KEY][CHARGER_SERIAL_NUMBER_KEY]}"
+
+    @property
+    def event(self) -> CalendarEvent | None:
+        """Return the last event."""
+        if not self.coordinator.data[CHARGER_LAST_EVENT]:
+            return None
+        return CalendarEvent(
+            summary=self.coordinator.data[CHARGER_LAST_EVENT].summary,
+            start=self.coordinator.data[CHARGER_LAST_EVENT].start,
+            end=self.coordinator.data[CHARGER_LAST_EVENT].end,
+            description=self.coordinator.data[CHARGER_LAST_EVENT].description,
+        )
+
+    async def async_get_events(
+        self, hass: HomeAssistant, start_date: datetime, end_date: datetime
```

Comment:
again, can we use WallboxEvent here? you can just `return await self.coordinator.async_get_sessions(...`

Reply:
Sadly not.  The previous remark from you is correct (def event(self) can return a WallboxEvent). But this one gives the pylint error Return type should be list[CalendarEvent] in async_get_events. 

I mainly based my changes on the radarr integration. And that one added a pylint exceprion (so it didn't check the pylint warning). But I got a a change requested from Joost, to remove exception (I completely understand)

Reply:
hmmm, see your point, alas, that means creating a calendarevent for each wallboxevent indeed

Reply:
let's do `        return [CalendarEvent(start=event.start, end = event.end, location=event.location, description=event.description, summary=event.summary) for event in await self.coordinator.async_get_sessions(
                self.coordinator.data[CHARGER_DATA_KEY][CHARGER_SERIAL_NUMBER_KEY],
                start_date,
                end_date,
            )]`